lessons/flow.md

```diff
@@ -0,0 +1,132 @@
+
+
+# Modern JS workshop - Flow
+
+## Table of contents
+1. [What is Flow?](#1-what-is-flow)
+2. [Getting started](#2-getting-started)
+   * [Setup the compiler](#setup-the-compiler)
+   * [Setup Flow client](#setup-flow-client)
+   * [Prepare your files for Flow](#prepare-your-files-for-flow) 
+3. [Writing Flow code](#3-writing-flow-code)
+4. [Where do we use it at The Guardian?](#4-where-do-we-use-it-at-the-guardian)
+5. [Resources](#5-resources)
+
+
+## 1. What is Flow?
+
+According to Flow's site:
+
+> Flow is a static type checker for javascript
+
+
+<! -- What does Flow give us? -->
+
+## 2. Getting started
+
+Let's start by cloning the repository in your local computer. Then, checkout the branch called `babel-solution`, this will be our starting point. In order to use Flow, you will need to setup the compiler in the project and install the client which runs the checker.
+
+### Setup the compiler
+
+The compiler is needed to **remove Flow types from the codebase**. It is possible to choose between `Babel` and a small CLI tool called `flow-remove-types`. 
+
+We are going to install it via Babel.
+
+#### 1. Install babel-preset-flow
+Install [`babel-preset-flow`](https://www.npmjs.com/package/babel-preset-flow) by executing the following `yarn` command:  
+
+```
+yarn add --dev @babel/preset-flow
+```
+
+If you are using `npm` the analogous command is:
+
+```
+npm install --dev @babel/preset-flow
```

Comment:
Should we try and encourage either npm or yarn for consistency? I don't think we should, but I was wondering it myself ...

Reply:
I would prefer `yarn` (because of that I put it first in the document). Nevertheless, I also added it the `npm` commands for the developer who uses `npm`.

Reply:
Yeah - I'll do the same then 👍